### PR TITLE
Add support for v2 of the SHARE API

### PR DESCRIPTION
**Pull request by dbrower**
This PR only supports the PushDocument part of the API, for uploading citations to SHARE. Left undone is the Search API. But we would like to merge this so others can build on it.

The new API is exposed as a new class, so existing code will continue to work (albeit, using the v1 API).

This is joint work with Huawei Weng at UCSD.

Comments and Discussion welcome.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -50,12 +54,21 @@ Style/CollectionMethods:
 Style/WordArray:
   Enabled: false
 
+Style/SymbolArray:
+  Enabled: false
```
Comment (dunn) [inline .rubocop.yml]: This could also be `EnforcedStyle: brackets` to enforce the old style.

File: .rubocop.yml [hunk 2]
```diff
@@ -50,12 +54,21 @@ Style/CollectionMethods:
...
+
 Style/RegexpLiteral:
   Enabled: false
 
 Style/StringLiterals:
   Enabled: false
 
+Style/FrozenStringLiteralComment:
+  Enabled: false
```
Comment (dunn) [inline .rubocop.yml]: These comments are easy enough to add with rubocop's `--auto-correct`.

File: .rubocop.yml (hunk 1 above)
Comment (dbrower) [reply .rubocop.yml]: Yes, that would probably be better.

File: .rubocop.yml (hunk 2 above)
Comment (dbrower) [reply .rubocop.yml]: We are using ruby < 2.3 in production, so it seemed better to keep the explicit `freeze`s. Maybe I should add a comment stating that.

**Comment by dbrower** [comment]
I push anychanges up on Wednesday.

File: .rubocop.yml (hunk 2 above)
Comment (dunn) [reply .rubocop.yml]: I haven't tried it but setting `TargetRubyVersion` might disable checks that require a newer Ruby http://rubocop.readthedocs.io/en/latest/configuration/#setting-the-target-ruby-version

Comment (mjgiarlo) [reply .rubocop.yml]: :+1:, we're targeting Ruby 2.1 in Hyrax FWIW.

**Comment by dbrower** [comment]
@dunn I have updated the rubocop file, and I set a ruby version of 2.1.

**Review by jeremyf — APPROVED** [review]

**Pull request merged**